crates/bevy_tasks/src/pollable_task.rs

```diff
@@ -0,0 +1,33 @@
+use crate::Task;
+use async_channel::{Receiver, TryRecvError};
+
+/// A pollable task whose result readiness can be checked in system functions
+/// on every frame update without blocking on a future
+#[derive(Debug)]
+pub struct PollableTask<T> {
+    receiver: Receiver<T>,
```

Comment:
Is using channels actually preferable to calling `future::block_on`? Has anyone run benchmarks? Will this work on `wasm`?  (pretty sure `future::block_on` doesn't, but its still worth checking)
Are there alternatives that don't involve async channels (which will allocate and use atomics)?

Reply:
That looks like https://github.com/bevyengine/bevy/pull/2650.

I haven't run the numbers on it though. We need to get data into the ecs from outside the ecs, which means that it needs to live somewhere with a stable address so that both sides can talk to each other. Currently, that's where the `Task` lives, in this PR it's in the channel internals.

Adding new items can also happen at any time relative to the system code happening, which means we definitely need atomics in some form.

Reply:
There were these benches from the original author, hanabi1224.

https://github.com/hanabi1224/bevy/blob/accee1bb4e1b3ea8c628199c0fd4ec02badc594f/examples/async_tasks/async_bench.rs

Looking at it, they probably should be ported to criterion and a bench needs to be added for the approach in this PR.

Reply:
I think another option is to use or implement something similar to `FuturesExt::now_or_never`. It would have almost no overhead compared to using `future::block_on`.

Reply:
I can confirm the example doesn't currently compile on wasm, since the single threaded task pool doesn't have the spawn_pollable api. You might be able to get it to work on wasm by detaching the task instead of storing it and then copying the code for spawn pollable over to the single threaded task pool.